### DISCUSSION TITLE
Script that can toggle RTX super resolution, since we can't use some filters with it.

### DISCUSSION THREAD

**Discussion by tms369** [mpv-player/mpv#15013] [Q&A]
I found out that we can't mirror flip when using RTX super resolution.
When I try to mirror flip the player it just says:

```
video filters:
d3d11vpp[scale=2 scaling mode=nvidia]
hflip[]
``` 
I guess the d3d11vpp filter takes over everything and won't let me use the flip filter. 
I am just searching for a simple script that enables d3d11vpp filter by default but can use a key to toggle it.( I can then apply mirror flip and then reapply d3d11vpp afterwards.)
Any inputs is appreciated.

**Comment by kasper93** [comment]
> I guess the d3d11vpp filter takes over everything and won't let me use the flip filter.

Not really specific to `d3d11vpp`. `hflip` filter doesn't work with hardware frames, and `d3d11vpp` requires hardware frames to work.

> I am just searching for a simple script that enables d3d11vpp filter by default but can use a key to toggle it.

you can toggle any filter with `change-list`. For example 
```
x change-list vf toggle 'hflip'
```
in your input.conf will toggle hflip filter with x key.

> I can then apply mirror flip and then reapply d3d11vpp afterwards.

Depending on the setup you can apply hflip in different ways. 

If you want to use hwdec, you can hflip after `d3d11vpp` by adding `vf=format=nv12` before hflip filter to download the frames to cpu.

If you don't use hwdec, you can apply hflip before `d3d11vpp`, but adding filter before.

The most optimal thou would be to use glsl shader to flip the image, instead of doing this on cpu. Will work with any combination of filters. you could use something like https://github.com/natural-harmonia-gropius/hdr-toys/blob/41d1044e64e3ba06de966c09b72e23b76690cf4a/shaders/hdr-toys/utils/transform.glsl to achieve that.

Or if you are using `gpu-next`, you can do `libplacebo-opts=distort=on,distort_scale_x=-1` for native flipping.
